pkg/query-service/ee/usage/manager.go

```diff
@@ -0,0 +1,292 @@
+package usage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/zap"
+
+	licenseserver "go.signoz.io/query-service/ee/integrations/signozio"
+	"go.signoz.io/query-service/ee/license"
+	"go.signoz.io/query-service/ee/model"
+	"go.signoz.io/query-service/utils/encryption"
+)
+
+// collect usage every hour
+var collectionFrequency = 1 * time.Hour
+
+// send usage every 24 hour
+var uploadFrequency = 24 * time.Hour
+
+const (
+	MAX_RETRIES            = 3
+	RETRY_INTERVAL_SECONDS = 5 * time.Second
+)
+
+type Manager struct {
+	repository *Repository
+
+	licenseRepo *license.Repo
+
+	// end the usage routine, this is important to gracefully
+	// stopping usage reporting and protect in-consistent updates
+	done chan struct{}
+
+	// terminated waits for the UsageExporter go routine to end
+	terminated chan struct{}
+}
+
+func StartManager(dbType string, db *sqlx.DB, licenseRepo *license.Repo, clickhouseConn clickhouse.Conn) (*Manager, error) {
+
+	repo := NewUsageRepo(db, clickhouseConn)
+	err := repo.InitDB(dbType)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to initiate usage repo: %v", err)
+	}
+
+	m := &Manager{
+		repository:  &repo,
+		licenseRepo: licenseRepo,
+	}
```

Comment:
You probably know this already but overlooked here. An idiomatic go-way would be create a manager object with `New` and then call `Start` on it which in turn calls the internal `start`. The `Start` should just start the manager instead of creating a new instance and returning it. 

Can start be called multiple times, assume happened for some hypothetical reason (maybe bug)? If not, then return an error in such case.

Reply:
Make sense to restrict start being called multiple times.